mesonbuild/backend/ninjabackend.py

```diff
@@ -3575,7 +3600,12 @@ def generate_ending(self):
             for t in deps.values():
                 # Add the first output of each target to the 'all' target so that
                 # they are all built
-                targetlist.append(os.path.join(self.get_target_dir(t), t.get_outputs()[0]))
+                #Add archive file if shared library in AIX for build all.
+                if mesonlib.is_aix():
+                    aix_tmp = re.sub ('[.][a]([.]?([0-9]+))*([.]?([a-z]+))*','.a', t.get_outputs()[0].replace ('.so','.a'))
+                    targetlist.append(os.path.join(self.get_target_dir(t), t.get_outputs()[0] if mesonlib.is_aix() and not(t.get_outputs()[0].endswith('.so')) else aix_tmp))
```

Comment:
you've already checked that you're on aix in this block, so you shouldn't need to check for aix again inside the block, right?

Reply:
This is done. Kindly check the latest commit